### PR TITLE
Drop support for Scala 2.10

### PR DESCRIPTION
**Pull request by sideeffffect**
/cc @sjrd

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ libraryDependencies += "org.portable-scala" %%% "portable-scala-reflect" % "1.0.
 
 `portable-scala-reflect` 1.0.0 supports:
 
-* Scala 2.10.x, 2.11.x, 2.12.x and 2.13.x
+* Scala 2.11.x, 2.12.x and 2.13.x
```
Comment (LeeTibbert) [inline README.md:25]: re: .x   

The .x is prior art, but isn't it a bit misleading here.  If I understand correctly
2.11.12 is supported/published, but, say, 2.11.0 is not.  Same story, next verse for 2.12 and 2.13.

portable-scala-crossproject (sp?) uses an idiom such as 2.12.10+.
(where I am taking the 2.12.10 from the .travis.yml above. similar
for 2.13 and 2.11, I believe).

Sorry if I am entirely off base here and/or that  I did not twig to this sooner.

Comment (sjrd) [reply README.md:25]: Scala Native is the only platform which is restricted to specific Scala versions. On Scala/JVM and Scala.js, any minor version can be used. So it is correct to say .x here.

**Review by sjrd — APPROVED** [review]
Thank you.

**Comment by SethTisue** [comment]
the day 2.10 finally died

**Pull request merged**